src/IECore/StreamIndexedIO.cpp

```diff
@@ -401,22 +402,17 @@ size_t compress(
 	int compressionLevel,
 	const std::string &compressor,
 	int threadCount,
-	boost::optional<size_t> maxBlockSize
+	boost::optional<size_t> maxBlockSize = boost::optional<size_t>(),
+	size_t minCompressedBlockSize = 1024U
 )
 {
 	size_t maxCompressedBlockSize = maxBlockSize ? maxBlockSize.get() : BLOSC_MAX_BUFFERSIZE;
-	constexpr size_t minCompressedBlockSize = 1 * 1024U;
 
 	if( size < minCompressedBlockSize )
 	{
 		return 0;
 	}
 
-	if ( compressionLevel == 0 )
```

Comment:
If I understand correctly : 

- When compression is off, we're now using Blosc to copy `data` uncompressed into `outputBuffer`
- In that case, `writeUniqueDataCompressed()` is still writing `data` and not `compressedBuffer` (because the size of the output buffer is not smaller).
- So we're just copying the data needlessly.

Hopefully I'm wrong?